### PR TITLE
DOM task

### PR DESCRIPTION
**Pull request by EnF1NeeT**
### DOM

[Demo](https://enf1neet.github.io/DOM-task/) |
[Code base](https://github.com/EnF1NeeT/DOM-task)

The code is submitted in a dedicated feature branch.

Only code files are submitted.

Please, review.

### CODE REVIEW

**Comment by EnF1NeeT** [comment]
Hi! Can I get feedback on my latest commit and in general on PR? :)

File: submissions/EnF1NeeT/DOM-task/script.js [hunk 1]
```diff
@@ -0,0 +1,155 @@
+const postsLst = [
+    {
+        id: '1',
+        topzoneImg:'./img/2.png',
+        p:'@Jonnie',
+        imgSrc: './img/post1.jpg',
+        imgSrc2: './img/post2.jpg',
+        imgSrc3: './img/post3.jpg',
+        imgSrc4: './img/post4.jpg'
```
Comment (madmaxWMFU) [inline submissions/EnF1NeeT/DOM-task/script.js:9]: Add a collection of photos in the array and set it to property imageList.

File: submissions/EnF1NeeT/DOM-task/script.js [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+btnContainer.addEventListener('click', addContent);
+
+function addContent({ target }) {
+    if (target.id) {
+        const contant = postsLst.find((pos) => pos.id === target.id);
+        mainContainer.innerHTML = post(contant);
+    }
+};
+
+function post({ imgSrc, imgSrc2, imgSrc3, imgSrc4, p, topzoneImg}) {
```
Comment (madmaxWMFU) [inline submissions/EnF1NeeT/DOM-task/script.js:53]: Set a more informative name for the function.

File: submissions/EnF1NeeT/DOM-task/script.js [hunk 3]
```diff
@@ -0,0 +1,155 @@
+const postsLst = [
```
Comment (madmaxWMFU) [inline submissions/EnF1NeeT/DOM-task/script.js:1]:
```suggestion
const postsList = [
```

File: submissions/EnF1NeeT/DOM-task/script.js [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+    {
+        id: '1',
+        topzoneImg:'./img/2.png',
+        p:'@Jonnie',
```
Comment (madmaxWMFU) [inline submissions/EnF1NeeT/DOM-task/script.js:5]: Maybe a nickname?

File: submissions/EnF1NeeT/DOM-task/script.js [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+];
+
+const btnContainer = document.querySelector('.friends');
+const mainContainer = document.querySelector('.posts');
+
+btnContainer.addEventListener('click', addContent);
+
+function addContent({ target }) {
+    if (target.id) {
+        const contant = postsLst.find((pos) => pos.id === target.id);
```
Comment (madmaxWMFU) [inline submissions/EnF1NeeT/DOM-task/script.js:48]: Set informative naming. Maybe named `targetContent` was be good.

Comment (madmaxWMFU) [inline submissions/EnF1NeeT/DOM-task/script.js:48]:
```suggestion
        const contant = postsList.find((post) => post.id === target.id);
```

**Review by madmaxWMFU — CHANGES_REQUESTED** [review]
Good work. See some comments below.
Check your demo on [validator](https://validator.w3.org/nu/?doc=https%3A%2F%2Fenf1neet.github.io%2FDOM-task%2F)

File: submissions/EnF1NeeT/DOM-task/script.js [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+const postsList = [
+>>>>>>> 2707486fe7b80d419dcc0d7f8c3d8c626d02b823
+    {
+        id: '1',
+        topzoneImg:'./img/2.png',
+        nickname:'@Jonnie',
+        imgSrc: './img/post1.jpg',
+        imgSrc2: './img/post2.jpg',
+        imgSrc3: './img/post3.jpg',
+        imgSrc4: './img/post4.jpg'
```
Comment (madmaxWMFU) [inline submissions/EnF1NeeT/DOM-task/script.js:15]: If the user has more than thousands of photos in his collection. Then you create a new property with a photo. I recommend saving the collection of photos into an array.

**Review by madmaxWMFU — CHANGES_REQUESTED** [review]

File: submissions/EnF1NeeT/DOM-task/script.js [hunk 7]
```diff
@@ -0,0 +1,169 @@
+<<<<<<< HEAD
+const postsList  = [
+||||||| 76728da
+const postsLst = [
+=======
+const postsList = [
+>>>>>>> 2707486fe7b80d419dcc0d7f8c3d8c626d02b823
```
Comment (madmaxWMFU) [inline submissions/EnF1NeeT/DOM-task/script.js:7]: After changing you don't delete pull messages

Comment (EnF1NeeT) [reply submissions/EnF1NeeT/DOM-task/script.js]: Done

File: submissions/EnF1NeeT/DOM-task/script.js (hunk 6 above)
Comment (EnF1NeeT) [reply submissions/EnF1NeeT/DOM-task/script.js]: Done

File: submissions/EnF1NeeT/DOM-task/script.js [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+                    <button class="btn">
+                        <img src="./img/send_FILL0_wght400_GRAD0_opsz48.png" alt="" class="btnImg">
+                        Share
+                    </button>
+                </div>
+                <div class="postTime">posted <time datetime="24.5.3222">posted 5h ago</time></div>
+                </article>`
+        };
+    return list;
+};
```
Comment (madmaxWMFU) [inline submissions/EnF1NeeT/DOM-task/script.js:84]: Add an empty line.

File: submissions/EnF1NeeT/DOM-task/script.js [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+                    </button>
+                    <button class="btn">
+                        <img src="./img/send_FILL0_wght400_GRAD0_opsz48.png" alt="" class="btnImg">
+                        Share
+                    </button>
+                </div>
+                <div class="postTime">posted <time datetime="24.5.3222">posted 5h ago</time></div>
+                </article>`
+        };
+    return list;
```
Comment (madmaxWMFU) [inline submissions/EnF1NeeT/DOM-task/script.js:83]:
```suggestion
    return imageList.map(image => {
        return `<article class="post">
                <div class="topzone">
                    <img class="topzoneImg" src="${topzoneImg}" alt="postImg">
                    <p>${nickname}</p>
                </div>
                <div class="postImg">
                    <img src="${image}" class="postImg" alt="postImg">
                </div>
                <div class="actionZone">
                    <button class="btn">
                        <img src="./img/favorite_FILL0_wght400_GRAD0_opsz48.png" alt="" class="btnImg">
                        Like
                    </button>
                    <button class="btn">
                        <img src="./img/forum_FILL0_wght400_GRAD0_opsz48.png" alt="" class="btnImg">
                        Comment
                    </button>
                    <button class="btn">
                        <img src="./img/send_FILL0_wght400_GRAD0_opsz48.png" alt="" class="btnImg">
                        Share
                    </button>
                </div>
                <div class="postTime">posted <time datetime="24.5.3222">posted 5h ago</time></div>
                </article>`; }).join('');
```

File: submissions/EnF1NeeT/DOM-task/script.js (hunk 8 above)
Comment (madmaxWMFU) [inline submissions/EnF1NeeT/DOM-task/script.js:84]: Add an empty line.

**Review by madmaxWMFU — CHANGES_REQUESTED** [review]

File: submissions/EnF1NeeT/DOM-task/script.js (hunk 9 above)
Comment (EnF1NeeT) [reply submissions/EnF1NeeT/DOM-task/script.js]: Done

File: submissions/EnF1NeeT/DOM-task/script.js (hunk 8 above)
Comment (EnF1NeeT) [reply submissions/EnF1NeeT/DOM-task/script.js]: Done

File: submissions/EnF1NeeT/DOM-task/script.js [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+
+function targetContent ({ target }) {
+    if (target.id) {
+        const contant = postsList.find((post) => post.id === target.id);
+        mainContainer.innerHTML = postGenerator(contant);
+    }
+};
+
+let list = '';
+function postGenerator({nickname, topzoneImg, imageList}, list) {
```
Comment (madmaxWMFU) [inline submissions/EnF1NeeT/DOM-task/script.js:54]:
```suggestion
function postGenerator({nickname, topzoneImg, imageList}) {
```

**Review by madmaxWMFU — CHANGES_REQUESTED** [review]

**Review by madmaxWMFU — APPROVED** [review]
Great, I approve of your work. Good luck with future tasks!

**Pull request merged**